src/Stripe.net/Infrastructure/ParameterBuilder.cs

```diff
@@ -33,6 +33,16 @@ public static string ApplyAllParameters(this StripeService service, object obj,
                             RequestStringBuilder.ProcessPlugins(ref requestString, attribute, property, value, obj);
                     }
                 }
+
+                var field = obj.GetType().GetRuntimeField("ExtraParams");
+                if (field != null)
+                {
+                    foreach (KeyValuePair<string, string> pair in (Dictionary<string, string>) field.GetValue(obj))
+                    {
+                        var key = WebUtility.UrlEncode(pair.Key);
+                        RequestStringBuilder.ApplyParameterToRequestString(ref requestString, key, pair.Value);
```

Comment:
Does this mean that you could pass `extra_params[sub_params][subsub_value]` and it would work?

Reply:
The brackets would be URL-encoded. I think the API doesn't mind and would still interpret the value as a nested object, but not entirely sure tbh. Maybe URL-encoding the key is not the right thing to do here, and instead we'd make it the caller's responsibility to pass a properly URL-encoded key?

Reply:
hmmm that seems frustrating for users to have to do this (url encoding themselves). Maybe not supporting sub-hashes for now is fine then. Also we could do a plugin that says "oh this is a dictionary, let's properly encode those"

Reply:
I'd rather keep the feature simple for now. If someone actually needs to use this method with nested objects we can think about it then.